src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -29,10 +29,27 @@ public RouteEntry(RouteTemplate template, Type handler, string[] unusedRoutePara
 
         internal void Match(RouteContext context)
         {
+            string? catchAllValue = null;
+
+            // If this template contains a catch-all parameter, we can concatenate the pathSegments
+            // at and beyond the catch-all segment's position. For example:
+            // Template:        /foo/bar/{*catchAll}
+            // PathSegments:    /foo/bar/one/two/three
+            if (Template.ContainsCatchAllSegment && context.Segments.Length >= Template.Segments.Length)
+            {
+                if(Template.Segments.Last().EncodeSlashes)
+                {
+                    catchAllValue = string.Join("%2F", context.Segments[Range.StartAt(Template.Segments.Length - 1)]);
+                }
+                else
+                {
+                    catchAllValue = string.Join('/', context.Segments[Range.StartAt(Template.Segments.Length - 1)]);
+                }
```

Comment:
I realized now that I overlooked this. We should be using Uri.UnescapeDataString here. It is likely that the previous behavior for `*` was to not do anything, which should have not been the case, as that's the behavior `**` has on ASP.NET Core routing.

I think I need to think about this a bit more, and chat with some other folks on the team, since this is a small breaking change, and I just want to make sure we are ok with it before we take it.

Reply:
I think previously `*` would be completely rejected in a template, right?  So for Blazor routing we aren't introducing a breaking change, to my knowledge.  I do understand the problem now, I seem to have had it backward in my head, so I can at least start by updating to unescape these segments while you double-check with the team.

Reply:
That's true, I had a brainfreeze :). So no breaking change, we should just make sure the behavior matches the one in ASP.NET Core routing, which is `*` decodes the url, and `**` does not

Reply:
Actually, `*` and `**` I believe are used for link generation, which might not apply here. Give me a second to confirm, it's been a while since I've looked at this specifically.

Reply:
Doh, `**` is used for link generation... so I've really mangled my interpretation of what needs to happen here.  In this case, I can just decode the entire catch-all value when it's incoming.  And I don't really need to look for `**` but just `*` presumably.

Reply:
It looks like I can simplify this by not worrying about `**`.  Also the `RouteContext` already decodes the values when adding the context segments so I believe that is taken care of as well.

Reply:
That makes sense. Lets just do `*` for now. We can always add `**` back if we have to

Reply:
Final question: do we want to `throw` if a template starts with more than one `*`?

Reply:
Yes, that way we make sure we can add it in the future without breaking anyone

Reply:
Thanks! That's what I thought, just wanted to confirm.